x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -123,6 +124,7 @@ else if (p.resolved()) {
 
                     // Do not fail multiple times in case the children are already unresolved.
                     if (ae.childrenResolved() == false) {
+                        checkFoldablesConvertFunction(ae, failures);
```

Comment:
#112668 is opened to follow up with this issue.

Reply:
I agree that the solution should be in the `Analyzer.ResolveUnionTypes`, and also think it could be related to another idea we have, which is to support union-types for more functions than just convert functions. Currently we assume that all convert functions support union-types and no other functions do. But this discussion clarifies that there exist convert functions that do not. So what about creating an interface for functions that support union-types, make sure all but these foldable functions use that, and this opens up the option to add union-types to other functions too?